infrastructure/tasks/10-functions.ts

```diff
@@ -19,46 +20,61 @@ import * as config from "../tfvars.json";
 import storageManagementClient = require("azure-arm-storage");
 import webSiteManagementClient = require("azure-arm-website");
 
+import AppInsights = require("azure-arm-appinsights");
 import CosmosDBManagementClient = require("azure-arm-cosmosdb");
 
 const getAppServicePlan = async (client: any) => {
   return await client.appServicePlans.get(
-    (config as any).azurerm_resource_group_00,
-    (config as any).azurerm_app_service_plan_00
+    (config as any).azurerm_resource_group,
+    (config as any).azurerm_app_service_plan
   );
 };
 
+const getAppInsightsKey = async (appInsightsClient: AppInsights) => {
+  const appInsightsInstance = await appInsightsClient.components.get(
+    (config as any).azurerm_resource_group,
+    (config as any).azurerm_application_insights
+  );
+  return appInsightsInstance.instrumentationKey;
+};
+
 export const run = async () => {
   const loginCreds = await login();
 
+  // Get AppInsights instrumentation key
+  const appInsightsClient = new AppInsights(
+    loginCreds.creds as any,
+    loginCreds.subscriptionId
+  );
+  const appInsightsKey = await getAppInsightsKey(appInsightsClient);
+
   // Needed to get storage connection string
   const storageClient = new storageManagementClient(
     loginCreds.creds as any,
     loginCreds.subscriptionId
   );
   const storageAccountKeys = await storageClient.storageAccounts.listKeys(
-    (config as any).azurerm_resource_group_00,
-    (config as any).azurerm_storage_account_00
+    (config as any).azurerm_resource_group,
+    (config as any).azurerm_storage_account
   );
   if (!storageAccountKeys || !storageAccountKeys.keys) {
     throw new Error("storageAccountKeys not found");
   }
   const storageConnectionString = `DefaultEndpointsProtocol=https;AccountName=${(config as any)
-    .azurerm_storage_account_00};AccountKey=${storageAccountKeys.keys[0]
-    .value}`;
+    .azurerm_storage_account};AccountKey=${storageAccountKeys.keys[0].value}`;
```

Comment:
what's happening here? this deserves a comment, what's supposed to be in `keys[0]` ?

Reply:
that's a field of the internal azure API, actually, the first of the key associated with the cosmosdb account (it's documented inside the azure api docs)

Reply:
there's the comment you should add to the code, with a link to the reference doc